options/options.js

```diff
@@ -85,9 +85,9 @@
       const backgroundPage = browser.extension.getBackgroundPage();
 
       backgroundPage.deleteProfile(id);
-    }
 
-    return deleteProfile(id);
+      return deleteProfile(id);
```

Comment:
I noticed that if you try to remove new profiles (doesn't have an id yet), this does essentially nothing. I moved the return inside the if statement to avoid an unnecessary promise and filter operations.

Reply:
For context

```javascript
  const deleteProfile = async (id) => {
    const storedProfiles = await getProfiles();
    const updatedProfiles = storedProfiles.filter(sp => sp.id !== id);
    return browser.storage.sync.set({ [PROFILES]: updatedProfiles });
  };
```
Delete profile just filter out any stored profile based on the id to remove.